pyasdf/asdf_data_set.py

```diff
@@ -1833,7 +1833,7 @@ def process(self, process_function, output_filename, tag_map,
         if self.mpi:
             self.mpi.comm.barrier()
 
-        if not self.mpi:
+        if not self.mpi and self.__compression[0] is not None:
```

Comment:
```python
if not self.mpi:
    compression = self.__compression[0]
```

I think this should have the same effect.

Reply:
This breaks the other multiprocessing test as compression would be set to "gzip", which is not one of the options in the COMPRESSIONS dict.

Reply:
Hmm - I think there must be some other underlying problem here. `("gzip", 2)` is (in the `COMPRESSIONS` mapping) mapped to `"gzip-2"` which should raise when passed to `h5py`.

I think https://github.com/SeismicData/pyasdf/blob/master/pyasdf/header.py#L38
should be changed to `COMPRESSIONS[value] = value`.